packages/myst-cli/src/build/utils/localArticleExport.ts

```diff
@@ -33,12 +33,13 @@ async function runExportAndWatch(
 ): Promise<ExportResults> {
   let results = await exportFn(session, $file, exportOptions, projectPath, clean);
   if (watch) {
+    const articleFiles = articlesWithFile(exportOptions.articles);
     const watchedFiles = new Set([
       $file,
-      ...exportOptions.articles,
-      ...exportOptions.articles
-        .map((article) => {
-          return selectors.selectFileDependencies(session.store.getState(), article);
+      ...articleFiles.map(({ file }) => file),
```

Comment:
Since `toc` is converted to articles first, this watch function will not trigger on changes to the `toc`; only to the articles it includes.

(However, this isn't really a regression. Even if articles are defined directly, adding a new article before did not add that article to the watch. Picking up on changes to the article list / toc is a separate feature.)